CHANGELOG.md

```diff
@@ -0,0 +1,20 @@
+## CHANGELOG
+
+***This document logs all changes to the core code or individual modules which cause a 
+variation in the behaviour of the software for users. This log will be used to write 
+out the final release notes at the moment of tagging a release.***
+
+- When adding content to this document, make sure to create a section for each module 
+if the changes only impact a single module and that section does not already exist in
+the document. When changes affect the entire software, make sure to add them in the 
+core section.
+
+- When possible please provide the number of the pull request(s) containing the 
+changes in the following format: PR #1234
+
+
+### Core
+
+
+### Module 
+#### module1
```

Comment:
There must already be some things that have been merged worth adding for a start? Otherwise the things that have been already merged into master will likely be missed when doing the release.

Reply:
yes but i will add them after merging this in a separate PR because we might have a discussion on each element and I want people submitting PRs to start contributing asap